mcrouter/lib/network/AsyncMcServer.cpp

```diff
@@ -365,7 +366,26 @@ class McServerThread {
       if (!server_.opts_.ports.empty()) {
         socket_.reset(new folly::AsyncServerSocket());
         for (auto port : server_.opts_.ports) {
-          socket_->bind(port);
+          if (server_.opts_.listenAddresses.empty()) {
+            socket_->bind(port);
+          } else {
+            std::vector<folly::IPAddress> ipAddresses;
+            for (auto listenAddress : server_.opts_.listenAddresses) {
+              auto maybeIp = folly::IPAddress::tryFromString(listenAddress);
+              checkLogic(maybeIp.hasValue(),
+                         "Invalid listen address: {}",
+                         listenAddress);
+              auto ip = maybeIp.value();
+              if (ip.isV4()) {
```

Comment:
can you replace the next 5 lines with just: `ipAddresses.push_back(std::move(ip));` ?